File: read.js
var async = require('async'),
    fs = require('fs'),
    path = require('path')
parentDir = '/root/SSLCert/cert'
const util = require('util');
//const parentDir = "/Users/stanislasgirard/Documents/Dev/SSLCert/cert"
const { Certificate, PrivateKey } = require('@fidm/x509')
const mysql = require('mysql');
var fs = require('fs');

var read = 0
var errorNB = 0
// First you need to create a connection to the db host: 'database.cppynzdwfotc.eu-west-3.rds.amazonaws.com',
const con = mysql.createConnection({
    host: 'localhost',
    user: 'admin',
    password: 'Stanley78!',
});
var query;


    con.connect( (err) => {
        if (err) {
            console.log('Error connecting to Db');
            throw err;
            return;
        }
        console.log('Connection established');
        query = util.promisify(con.query).bind(con);
        var files = fs.readdirSync('/root/SSLCert/cert');
        processArray(files)
    
        /*var sql = "CREATE TABLE IF NOT EXISTS `Certificates`.`certificate` (`id` INT(11) NOT NULL AUTO_INCREMENT PRIMARY KEY, `domain` VARCHAR(255) NULL, `certificate` VARCHAR(16000) NULL);";
        con.query(sql, function(err, result) {
            if (err) throw err;
            console.log("Table created");
        });*/
    });  
    





async function processArray(files) {
    
    console.log("Going in")
    for (const file of files) {
        var filePath = path.join(parentDir, file);
        var cert = Certificate.fromPEM(fs.readFileSync(filePath))
        var CN = con.escape(cert.subject.commonName)
        var ON= con.escape(cert.subject.organizationName)
        var PKAlgo = con.escape(cert.publicKey.algo)
        var PK= con.escape(cert.publicKeyRaw)
        var KU = con.escape(cert.keyUsage)
        var ION = con.escape(cert.issuer.organizationName)
        var VF = con.escape(cert.validFrom)
        var VT = con.escape(cert.validTo)
        var sql = "INSERT INTO Certificates.decoded (filename, subjectCN, subjectON, pubkeyalgo, pubkey, issuerON, validFrom, validTo)" 
        sql += " VALUES ('" + file + "'," + CN + "," + ON +"," + PKAlgo + "," + PK + "," + ION + "," + VF + "," + VT + ");"
        
         query(sql, function(err, result) { 
            if (err){
                errorNB += 1
                    console.log("Error:", errorNB)
                    
                
            } else {
                read += 1
                if (read % 100 == 0){
                    console.log(read)
                }
            }
            
            return result
            
            
        })
        
        
    
    }
}

File: read.py
import os
path = '/root/SSLCert/cert/'
#path = '/Users/stanislasgirard/Documents/Dev/GetCertificates/certexample/'
import mysql.connector
from cryptography import x509
from cryptography.hazmat.backends import default_backend
from cryptography.hazmat.backends import default_backend
from cryptography.x509.oid import NameOID, ExtensionOID, ExtendedKeyUsageOID
from cryptography.hazmat.primitives.asymmetric import rsa

from mysql.connector import errorcode
number = 0
try:
    cnx = mysql.connector.connect(user='admin', password='Stanley78!',
                              host='localhost',
                              database='Certificates')
    cursor = cnx.cursor()
    for filename in os.listdir(path):
        
        with open(path + filename, 'rb') as content_file:
          try: 
            content = content_file.read()
            cert = x509.load_pem_x509_certificate(content, default_backend())
            issuer = 'undefined'
            subjectCN = 'undefined'
            subjectON = 'undefined'
            publicKey = 'not RSA'
            publicKeye = "0"
            publicKeyn = "0"
            keySize = "0"
            # Get Issuer
            if cert.issuer.get_attributes_for_oid(NameOID.ORGANIZATION_NAME):
                issuer = cert.issuer.get_attributes_for_oid(NameOID.ORGANIZATION_NAME)[0].value
            
            #Get Subject Common Name
            if cert.subject.get_attributes_for_oid(NameOID.COMMON_NAME):
                subjectCN = cert.subject.get_attributes_for_oid(NameOID.COMMON_NAME)[0].value
            

            #Get Key Size
            if cert.public_key:
                keySize = cert.public_key().key_size

            if isinstance(cert.public_key(), rsa.RSAPublicKey):
                publicKey = cert.public_key().public_numbers()
                publicKeye = publicKey.e
                publicKeyn = publicKey.n


            sql = "INSERT INTO Certificates.decoded (filename, issuerON, subjectCN, pubkeye, pubkeyn)  VALUES (%s, %s, %s, %s, %s);" 
            
            print(publicKeyn)
            result = cursor.execute(sql, (filename, issuer, subjectCN, publicKeye, publicKeyn))
            cnx.commit()
            number += 1
            if number % 100 == 0:
              print(number)
          except mysql.connector.Error as err:
            print(err)
          except:
            print("Error") 
            
            
            
            
            
        
    cursor.close()
    cnx.close()
except mysql.connector.Error as err:
  if err.errno == errorcode.ER_ACCESS_DENIED_ERROR:
    print("Something is wrong with your user name or password")
  elif err.errno == errorcode.ER_BAD_DB_ERROR:
    print("Database does not exist")
  else:
    print(err)





